pkg/karmadactl/get.go

```diff
@@ -123,62 +119,64 @@ type CommandGetOptions struct {
 }
 
 // NewCommandGetOptions returns a GetOptions with default chunk size 500.
-func NewCommandGetOptions(parent string, streams genericclioptions.IOStreams) *CommandGetOptions {
-	return &CommandGetOptions{
-		PrintFlags: get.NewGetPrintFlags(),
-
-		CmdParent: parent,
-
-		IOStreams:   streams,
-		ChunkSize:   500,
-		ServerPrint: true,
-	}
-}
-
-// Complete takes the command arguments and infers any remaining options.
-func (g *CommandGetOptions) Complete(cmd *cobra.Command, args []string) error {
+func NewCommandGetOptions() *CommandGetOptions {
 	newScheme := gclient.NewSchema()
-	// human readable printers have special conversion rules, so we determine if we're using one.
-	g.IsHumanReadablePrinter = true
+	kubeConfig := defaultKubeConfig
+	printFlags := get.NewGetPrintFlags()
+
+	ioStreams := genericclioptions.IOStreams{In: getIn, Out: getOut, ErrOut: getErr}
 
-	// check karmada config path
+	// using env variable if not empty
 	env := os.Getenv("KUBECONFIG")
 	if env != "" {
-		g.KubeConfig = env
+		kubeConfig = env
```

Comment:
Perhaps, we don't need to check env `KUBECONFIG`,  none of other subcommands check this env, but they can run. So I think we can optimize it here.

Reply:
this is need, or we'll need to emphasize the kubeconfig when executing the command, like below.
./karmadactl get ns --kubeconfig=/root/.kube/karmada.config